### PR TITLE
EDX-513: AT: EDX-247: DIS - View School Contacts

### PR DESCRIPTION
**Pull request by TimOsse**

### CODE REVIEW

File: test/src/services/institute-api-service.js [hunk 1]
```diff
@@ -38,6 +45,8 @@ const instituteApiService = {
     };
     const url = `${constants.institute_base_url}${SCHOOL_ENDPOINT}/paginated`;
     const userSchoolResult = await restUtils.getData(token, url, schoolSearchParam);
+    console.log('SCHOOL_RESULT:== ');
+    console.log(userSchoolResult);
```
Comment (SoLetsDev) [inline test/src/services/institute-api-service.js:49]: I'm not sure if you meant to use console.log here. Usually when we want to log something, we use log.info in tests. 

Could also use a log.debug if you dont' always want to show the school results.

File: test/src/test_cases/edx-districts-frontend/test-school-contacts.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+test('testPage', async t => {
+    // navigate to /schools, expect title
+    await t.navigateTo(base_url + '/schools');
+    await schools.clickSchoolContactsButton();
+    await t.expect(districtContacts.navTitle.innerText).contains('School Contacts');
+
+    //Verify the Principal and data is loaded
+    let schoolPrincipalContact = await getSchoolPrincipalDetails('03005');
+    console.log(schoolPrincipalContact);
```
Comment (SoLetsDev) [inline test/src/test_cases/edx-districts-frontend/test-school-contacts.js:49]: we can use log.info(`principle contact found ${name}`);

File: test/src/helpers/school-set-up-utils.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        return await restUtils.getData(token, url);
+    },
+    async getSchoolPrincipalDetails(schoolNumber){
+        let schoolPrincipal = '';
+        let schoolDetails = await schoolSetUpUtils.getSchoolDetails(schoolNumber);
+        for (const schoolContact of schoolDetails.contacts){
+            if(schoolContact.schoolContactTypeCode === 'PRINCIPAL'){
+                let parsedExpiryDate = null;
+                if (schoolContact.expiryDate) {
+                    parsedExpiryDate = new LocalDate.parse(schoolContact.expiryDate, DateTimeFormatter.ofPattern('uuuu-MM-dd\'T\'HH:mm:ss'));
```
Comment (SoLetsDev) [inline test/src/helpers/school-set-up-utils.js:25]: after yesterday's date fiasco, we probably want to use LocalDateTime. LocalDate has funny behaviors with timestamps.  HH:mm:ss

File: test/src/helpers/school-set-up-utils.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                }
+                if (parsedExpiryDate === null) {
+                    schoolPrincipal = schoolContact;
+                }
```
Comment (SoLetsDev) [inline test/src/helpers/school-set-up-utils.js:29]: After looking a bit more into this function, could we confirm the logic here. It sounds like if a schoolPrincipal has an expiry date we do not send it back. 

I think your tactic here was to not send back principals that are expired. 

So I think we are missing a check to see if the expiry date has less than today. You can refer to here for the logic. https://github.com/bcgov/EDUC-STUDENT-ADMIN/blob/master/frontend/src/utils/institute/status.js 

Like where your heads at :) thinking of the edge cases.

File: test/src/page_models/schools-page.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Selector, t } from 'testcafe';
+import log from 'npmlog';
+
+class SchoolsPage {
+    constructor() {
+        this.navTitle = Selector('#navTitle');
```
Comment (SoLetsDev) [inline test/src/page_models/schools-page.js:6]: We can remove this line. Since it's covered in navBar.

**Review by SoLetsDev — APPROVED** [review]
